File: DEMO/DEMO/SQLiteManger.m
//
//  SQLiteManger.m
//  DEMO
//
//  Created by 我演示 on 2019/10/15.
//  Copyright © 2019 Jerry-Fu. All rights reserved.
//

#import "SQLiteManger.h"

@implementation SQLiteManger
static id _instance ;

+(instancetype)shareInstance {

    
    static dispatch_once_t  onceToken;
    dispatch_once(&onceToken, ^{
        _instance = [[self alloc] init];
    });
    return _instance;
    
}

-(BOOL)openDB {
    /// 数据库文件存放的路径
    NSString *filePath = [NSSearchPathForDirectoriesInDomains(NSDocumentDirectory, NSUserDomainMask, YES) firstObject] ;
    
    filePath = [filePath stringByAppendingPathComponent:@"demo.sqlite"] ;
    
    /// 打开一个数据库类型
//    sqlite3

    return  YES;
}
-(BOOL)createTable {
    
    
    return  YES;
}

@end

File: Swift信息/Swift信息/SQLiteManger.swift
//
//  SQLiteManger.swift
//  Swift信息
//
//  Created by 我演示 on 2019/10/15.
//  Copyright © 2019 Jerry-Fu. All rights reserved.
//

import UIKit


class SQLiteManger: NSObject {

    ///1: 创建类的实例变量
    static let instance = SQLiteManger()
    
    /// let 是线程安全的
    
    ///2: 对外提供单利的接口
    
    class func shareInstance() {
        
    }
 
    func openDB() -> Bool {
        ///1: 封装创建表sql语句
        
        ///2: 执行sql语句
        
        return true
    }
    
}
